internal/transform/transaction.go

```diff
@@ -147,6 +150,20 @@ func TransformTransaction(transaction ingest.LedgerTransaction, lhe xdr.LedgerHe
 			outputSorobanResourcesInstructions = uint32(sorobanData.Resources.Instructions)
 			outputSorobanResourcesReadBytes = uint32(sorobanData.Resources.ReadBytes)
 			outputSorobanResourcesWriteBytes = uint32(sorobanData.Resources.WriteBytes)
+			outputInclusionFeeBid = int64(transactionEnvelopeV1.Tx.Fee) - outputResourceFee
+
+			accountBalanceStart, accountBalanceEnd := getAccountBalanceFromLedgerEntryChanges(transaction.FeeChanges, sourceAccount.Address())
+			initialFeeCharged := accountBalanceStart - accountBalanceEnd
+			outputInclusionFeeCharged = initialFeeCharged - outputResourceFee
+
+			meta, ok := transaction.UnsafeMeta.GetV3()
+			if ok {
+				accountBalanceStart, accountBalanceEnd := getAccountBalanceFromLedgerEntryChanges(meta.TxChangesAfter, sourceAccount.Address())
+				outputResourceFeeRefund = accountBalanceEnd - accountBalanceStart
+			}
+
+			// TODO: FeeCharged is calculated incorrectly in protocol 20. Remove when protocol is updated and the bug is fixed
+			outputFeeCharged = outputFeeCharged - outputResourceFeeRefund
```

Comment:
Just confirming that this won't have a negative impact on classic transactions since the `outputResourceFeeRefund` should always be 0 for classic, correct?

Reply:
Yeah this should work for classic. This is wrapped within a soroban only check for everything in this code block

Reply:
ah i missed that part, thanks